pkg/liqonet/natMappingInflater.go

```diff
@@ -0,0 +1,208 @@
+	}
+	if err == nil && natMappings != nil {
+		// Resource already exists
+		return nil
+	}
+	// Resource does not exist yet
+	res := &netv1alpha1.NatMapping{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "net.liqo.io/v1alpha1",
+			Kind:       "NatMapping",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: natMappingPrefix,
+			Labels: map[string]string{
+				"net.liqo.io/natmapping":     "true",
+				liqoconst.ClusterIDLabelName: clusterID,
+			},
+		},
+		Spec: netv1alpha1.NatMappingSpec{
+			ClusterID:    clusterID,
+			PodCIDR:      podCIDR,
+			ExternalCIDR: externalCIDR,
+			Mappings:     make(map[string]string),
+		},
+	}
+	unstructuredResource, err := runtime.DefaultUnstructuredConverter.ToUnstructured(res)
+	if err != nil {
+		klog.Errorf("cannot map resource to unstructured resource: %s", err.Error())
+		return err
+	}
+	// Create resource
+	up, err := inflater.dynClient.
+		Resource(netv1alpha1.NatMappingGroupResource).
+		Create(context.Background(), &unstructured.Unstructured{Object: unstructuredResource}, metav1.CreateOptions{})
+	if err != nil {
+		klog.Errorf("cannot create natMapping resource: %s", err.Error())
+		return err
+	}
+	klog.Infof("Resource %s for cluster %s successfully created", up.GetName(), clusterID)
+	return nil
+}
+
+// TerminateNatMappings deletes the NatMapping resource for remote cluster.
+func (inflater *NatMappingInflater) TerminateNatMappings(clusterID string) error {
+	// Get resource for remote cluster
+	natMappings, err := inflater.getNatMappings(clusterID)
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("cannot retrieve natMapping resource for cluster %s: %w", clusterID, err)
+	}
+	if err != nil && errors.IsNotFound(err) {
+		return nil
+	}
+	// Delete resource
+	err = inflater.dynClient.Resource(netv1alpha1.NatMappingGroupResource).Delete(
+		context.Background(), natMappings.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	klog.Infof("NatMapping resource for cluster %s deleted", clusterID)
+	return nil
+}
+
+// AddMapping adds a mapping in the resource related to a remote cluster.
+func (inflater *NatMappingInflater) AddMapping(oldIP, newIP, clusterID string) error {
+	// Get resource for remote cluster
+	natMappings, err := inflater.getNatMappings(clusterID)
+	if err != nil {
+		return fmt.Errorf("cannot retrieve natMapping resource for cluster %s: %w", clusterID, err)
+	}
+	// Check if mapping already exists
+	if value, exists := natMappings.Spec.Mappings[oldIP]; exists && value == newIP {
+		return nil
+	}
+	// Mapping does not exist yet
+	natMappings.Spec.Mappings[oldIP] = newIP
+	// Update resource
+	if err := inflater.updateNatMappings(natMappings); err != nil {
+		return fmt.Errorf("cannot update natMapping resource for cluster %s: %w", clusterID, err)
+	}
+	return nil
+}
+
+// RemoveMapping deletes a mapping from the resource related to a remote cluster.
+func (inflater *NatMappingInflater) RemoveMapping(oldIP, clusterID string) error {
```

Comment:
I was wondering: Would be useful to model the return values (bool, error) as @alacuku did in #652 to know if an effective change has been triggered and not just the absence of errors? @alacuku @davidefalcone1 what do you think?